module3/lessons/mobile_friendly_design_bootstrap.md

```diff
@@ -0,0 +1,208 @@
+
+
+### Bootstrap Classes - Columns
+Taking this math into account, Bootstrap gives us handy classes we can use to insert into our HTML to achieve the columns quickly and easily: 
+
+```html
+<div class="col-xs-12 col-sm-3">
+   Navigation
+</div>
+<div class="col-xs-12 col-sm-6">
+   Main Content
+</div>
+<div class="col-xs-12 col-sm-3">
+   Sidebar
+</div>
+```
+If we break these class names down: 
+* `col` specifies a column
+* `xs` specifies the screen size - `xs` for extra small (most phones in vertical orientation), `sm` for small (most phones in horizontal orientation), `md` medium (most tablets in vertical orientation), `lg` large (most laptop or computer screens), and `xl` for extra-large screens.
+* `[number]` specifies the number of columns wide this area should contain
+
+
+### Rows
+
+So far we've been focusing on the vertical elements - columns - but there are also __horizontal__ areas that Bootstrap calls **rows**. A row is another type of container element that organizes information horizontally, inside of a vertical column. 
+
+```html
+<div class="row">
+   <p>Some content...</p>
+</div>
+```
+
+In Bootstrap's CSS, a row should be used as a containing element for more columns, since it works to offset the CSS margin property of each column. 
+
+Example - pay close attention to this math! 
+
+```html
+<div class="row">
+   <!-- my first row -->
+   <div class="col-xs-12 col-sm-4 col-md-6">
+      <p>First column; on tiny screens it's full width, on small screens it's 1/3 of the width, and on medium screens and bigger it's half the width. </p>
+   </div>
+   <div class="col-xs-12 col-sm-4 col-md-6">
+      <p>Second column; similar to the above.</p>
+   </div>
+   <div class="col-xs-12 col-sm-4 col-md-12">
+      <p>Third column, but only appears as such on small screens! On both tiny and medium+ screens, it takes up the entire row of 12.</p>
+   </div>
+</div>
+<div class="row">
+   <!-- my second row - this content will never interfere with the first row -->
+   <div class="col-xs-12">
+      <p>I only have one column down here.</p>
+   </div>
+</div>
+```
+
+Technically, we could have a row of columns **inside of** a row of columns! This can sometimes get pretty messy, but it's possible. 
+
+In general, keep your layouts as clean & uncomplicated as possible - your users will thank you!
+
+
+
+
+## Practice
+
+1. Go to [the BBC.com news website](https://bbc.co.uk). On a piece of paper, sketch out the general _rows & columns_ that fit this site's design the best. (Note: news changes often, you may want to take a screenshot as reference.)
+2. Bootstrap's classes can be implemented even in a simple HTML document - Rails not required. 
+   1. Create a file called `bootstrap_practice.html`. Include the CDN files via the Bootstrap [Quick Start](https://getbootstrap.com/docs/5.3/getting-started/introduction/#quick-start) instructions. 
+   2. Take your sketched-out design from before and implement basic areas with basic content. 
```

Comment:
Could we elaborate a little more about the 'basic content' here? I was thinking something along the lines of
```
2. Take your sketched-out design from before and implement basic areas with basic content (such as some placeholder text inside of a `div` element).
```